### PR TITLE
[Doc] Update external.md

### PR DESCRIPTION
**Pull request by neozhaoliang**

### CODE REVIEW

File: docs/lang/articles/basic/external.md [hunk 1]
```diff
@@ -227,3 +227,19 @@ copy(x, y) # error!
 copy(x, y.clone()) # correct
 copy(x, y.contiguous()) # correct
 ```
+
+## Can I use `@ti.kernel` to accelerate a NumPy function?
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:231]:
```suggestion
## FAQ

### Can I use `@ti.kernel` to accelerate a NumPy function?
```

File: docs/lang/articles/basic/external.md [hunk 2]
```diff
@@ -227,3 +227,19 @@ copy(x, y) # error!
...
+
+Users frequently ask whether they can call NumPy functions to process an array inside the Taichi scope. For example, like the follows:
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:233]:
```suggestion
You may try to call NumPy functions to process an array inside the Taichi scope, as the following example shows:
```

File: docs/lang/articles/basic/external.md [hunk 3]
```diff
@@ -227,3 +227,19 @@ copy(x, y) # error!
...
+import numpy as np
+
+@ti.kernel
+def invalid_sum(arr: ti.types.ndarray()) -> float:
+    return np.sum(arr)
+```
+
+This is *not allowed*: Taichi doesn't compile NumPy's functions. This is a major difference between Taichi and other Python accelerating frameworks like Numba.
+
+If you want to use a Numpy function which Taichi does not offer its counterpart functionality, you can call it in the Python scope as usual, and pass the result array  to Taichi kernels via `ti.types.ndarray()`.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:245]:
```suggestion
If you want to use a NumPy function to complement Taichi, you can call the function in the Python scope as usual and pass the generated array to Taichi kernels via `ti.types.ndarray()`.
```

File: docs/lang/articles/basic/external.md [hunk 4]
```diff
@@ -227,3 +227,19 @@ copy(x, y) # error!
...
+
+```python
+import numpy as np
+
+@ti.kernel
+def invalid_sum(arr: ti.types.ndarray()) -> float:
+    return np.sum(arr)
+```
+
+This is *not allowed*: Taichi doesn't compile NumPy's functions. This is a major difference between Taichi and other Python accelerating frameworks like Numba.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:243]:
```suggestion
This is *not allowed*. Unlike other Python acceleration frameworks, such as Numba, Taichi does not compile NumPy functions. 
```

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

File: docs/lang/articles/basic/external.md (hunk 4 above)
Comment (yuanming-hu) [reply docs/lang/articles/basic/external.md:243]: *not allowed* -> *not supported* sounds more friendly.

File: docs/lang/articles/basic/external.md (hunk 3 above)
Comment (yuanming-hu) [reply docs/lang/articles/basic/external.md:245]: @neozhaoliang could you provide a code example here?

File: docs/lang/articles/basic/external.md [hunk 5]
```diff
@@ -227,3 +227,31 @@ copy(x, y) # error!
...
+
+@ti.kernel
+def invalid_sum(arr: ti.types.ndarray()):
+    total = np.sum(arr)
+    ...
+```
+
+This is *not supported*: Taichi doesn't compile NumPy's functions. This is a major difference between Taichi and other Python accelerating frameworks like Numba.
+
+If you want to use a Numpy function which Taichi does not offer its counterpart functionality, you can call it in the Python scope as usual, and pass the result array  to Taichi kernels via `ti.types.ndarray()`. For example:
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:248]:
```suggestion
If you want to use a NumPy function, which lacks a counterpart in Taichi, you can call the function in the Python scope as usual and pass the processed array to Taichi kernels via `ti.types.ndarray()`. For example:
```

File: docs/lang/articles/basic/external.md [hunk 6]
```diff
@@ -227,3 +227,31 @@ copy(x, y) # error!
...
+```python
+import numpy as np
+
+@ti.kernel
+def invalid_sum(arr: ti.types.ndarray()):
+    total = np.sum(arr)
+    ...
+```
+
+This is *not supported*: Taichi doesn't compile NumPy's functions. This is a major difference between Taichi and other Python accelerating frameworks like Numba.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:246]:
```suggestion
This is *not supported*. Unlike other Python acceleration frameworks, such as Numba, Taichi does not compile NumPy functions. 
```

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

**Review by yuanming-hu — APPROVED** [review]
LGTM, thanks!

File: docs/lang/articles/basic/external.md [hunk 7]
```diff
@@ -227,3 +227,31 @@ copy(x, y) # error!
 copy(x, y.clone()) # correct
 copy(x, y.contiguous()) # correct
 ```
+
+## FAQ
+
+### Can I use `@ti.kernel` to accelerate a NumPy function?
+
+You may try to call NumPy functions to process an array inside the Taichi scope, as the following example shows:
```
Comment (ailzhang) [inline docs/lang/articles/basic/external.md:235]:
```suggestion
Unlike other Python acceleration frameworks, such as Numba, Taichi does not compile NumPy functions. Calling NumPy functions inside the Taichi scope is not supported, as the following example shows:
```

File: docs/lang/articles/basic/external.md [hunk 8]
```diff
@@ -227,3 +227,31 @@ copy(x, y) # error!
...
+
+```python
+import numpy as np
+
+@ti.kernel
+def invalid_sum(arr: ti.types.ndarray()):
+    total = np.sum(arr)
+    ...
```
Comment (ailzhang) [inline docs/lang/articles/basic/external.md:243]: We should put the error message in the comment so that it's obvious non-supported case.

**Review by ailzhang — APPROVED** [review]
Thanks!

**Pull request merged**